sled-agent/src/sim/sled_agent.rs

```diff
@@ -194,6 +194,13 @@ impl SledAgent {
         initial_hardware: InstanceHardware,
         target: InstanceRuntimeStateRequested,
     ) -> Result<InstanceRuntimeState, Error> {
+        let ncpus: i64 = (&initial_hardware.runtime.ncpus).into();
+        if ncpus > 16 {
+            return Err(Error::internal_error(
```

Comment:
I know this should probably be `bad_request` instead, but for `test_instance_fails_to_boot_with_disk` I needed it to be an `internal_error`. I welcome any input for better ways to do this  :)

Reply:
First thought was `InvalidValue` but that also just gets mapped to 400 as well :/

Either way, add a comment here to indicate why this specific error

Reply:
added comments in be9f7a37